cmd/aws-iam-authenticator/server.go

```diff
@@ -33,13 +41,42 @@ var serverCmd = &cobra.Command{
 	Short: "Run a webhook validation server suitable that validates tokens using AWS IAM",
 	Long:  ``,
 	Run: func(cmd *cobra.Command, args []string) {
-		config, err := getConfig()
+		stopCh := signals.SetupSignalHandler()
 
+		config, err := getConfig()
 		if err != nil {
 			logrus.Fatalf("%s", err)
 		}
 
-		server.New(config).Run()
+		k8sconfig, err := clientcmd.BuildConfigFromFlags(config.Master, config.Kubeconfig)
+		if err != nil {
+			logrus.WithError(err).Fatal("can't create kubernetes config")
+		}
+
+		kubeClient, err := kubernetes.NewForConfig(k8sconfig)
+		if err != nil {
+			logrus.WithError(err).Fatal("can't create kubernetes client")
+		}
+
+		iamClient, err := clientset.NewForConfig(k8sconfig)
+		if err != nil {
+			logrus.WithError(err).Fatal("can't create iam authenticator client")
+		}
+
+		iamInformerFactory := informers.NewSharedInformerFactory(iamClient, time.Second*30)
```

Comment:
30s is quite low and means excessive apiserver traffic, e.g. controller-runtime defaults to 10hrs.

Reply:
Great point!